funcframework/framework.go

```diff
@@ -206,6 +215,62 @@ func wrapEventFunction(fn interface{}) (http.Handler, error) {
 	}), nil
 }
 
+func wrapTypedFunction(fn interface{}) (http.Handler, error) {
+	inputType, err := validateTypedFunction(fn)
+	if err != nil {
+		return nil, err
+	}
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		body, err := readHTTPRequestBody(r)
+		if err != nil {
+			writeHTTPErrorResponse(w, http.StatusBadRequest, crashStatus, fmt.Sprintf("%v", err))
+			return
+		}
+		argVal := inputType
+
+		if err := json.Unmarshal(body, argVal.Interface()); err != nil {
+			writeHTTPErrorResponse(w, http.StatusBadRequest, crashStatus, fmt.Sprintf("Error: %s, while converting input type data: %s", err.Error(), string(body)))
+			return
+		}
+
+		defer recoverPanic(w, "user function execution")
+		funcReturn := reflect.ValueOf(fn).Call([]reflect.Value{
+			argVal.Elem(),
+		})
+
+		handleTypedReturn(funcReturn, w)
+	}), nil
+}
+
+func handleTypedReturn(funcReturn []reflect.Value, w http.ResponseWriter) {
+	if len(funcReturn) == 0 {
+		return
+	}
+	errorVal := funcReturn[len(funcReturn)-1].Interface() // last return must be of type error
+	firstVal := funcReturn[0].Interface()
+	if errorVal != nil && reflect.TypeOf(errorVal).AssignableTo(errorType) {
+		writeHTTPErrorResponse(w, http.StatusInternalServerError, errorStatus, fmtFunctionError(errorVal))
+	} else if !reflect.TypeOf(firstVal).AssignableTo(errorType) {
+		returnVal, _ := json.Marshal(firstVal)
+		fmt.Fprintf(w, string(returnVal))
+	}
+}
+
+func validateTypedFunction(fn interface{}) (*reflect.Value, error) {
+	ft := reflect.TypeOf(fn)
+	if ft.NumIn() != 1 {
+		return nil, fmt.Errorf("expected function to have one parameters, found %d", ft.NumIn())
+	}
+	if ft.NumOut() > 2 {
+		return nil, fmt.Errorf("expected function to have maximum two return values")
+	}
+	if ft.NumOut() > 0 && !ft.Out(ft.NumOut()-1).AssignableTo(errorType) {
+		return nil, fmt.Errorf("expected last return type to be of error")
+	}
+	var inputType = reflect.New(ft.In(0))
```

Comment:
[nit] can this be inlined with the return statement?

Reply:
Tried that. This &(reflect.New(ft.In(0))) does not work